### ISSUE TITLE
BaseFinetuning freeze method doesn't freeze track_running_stats in BatchNorm

### ISSUE THREAD

**Issue by PososikTeam** [Lightning-AI/lightning#15062]
### Bug description

Continue of topic https://github.com/Lightning-AI/lightning/issues/13902.
Even with 
```
class CustomBackboneFineTuning(BackboneFinetuning):
    def freeze_before_training(self, pl_module):
        self.freeze(pl_module.backbone, train_bn=False)


trainer = Trainer(..., callbacks=[CustomBackboneFineTuning()])```

BatchNorm still change `running_mean` and `running_var`. So it's not expected behavior, because when you evaluate the model after training with `train_bn=False`, BatchNorm before and after training will have different result due to `running_mean` and `running_var` change.

### How to reproduce the bug

```python
Just train any model with 

class CustomBackboneFineTuning(BackboneFinetuning):
    def freeze_before_training(self, pl_module):
        self.freeze(pl_module.backbone, train_bn=False)


trainer = Trainer(..., callbacks=[CustomBackboneFineTuning()])```
And compare evaluated backbone results before and after training.
```

### Error messages and logs

```

# Error messages and logs here please

```

### Environment

```

#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):

```
